demo/src/components.rs

```diff
@@ -8,8 +8,8 @@ use rafx::visibility::DynamicAabbVisibilityNodeHandle;
 
 #[derive(Clone)]
 pub struct MeshComponent {
-    pub render_node: MeshRenderNodeHandle,
-    pub visibility_node: DynamicAabbVisibilityNodeHandle,
+    pub render_node: Option<MeshRenderNodeHandle>,
```

Comment:
these are now populated during the extract phase. this does break the general convention that extract should be read-only over all data so that we can do it in parallel across all data. not sure what to do about it

Reply:
We discussed this on discord and considered a few options:
1) Rely on jobs that write to the same components do not run at the same time (this is checked at runtime)
2) Use interior mutability. Probably not better than option 1 because it just silently hides the write and could result in inconsistent logical behavior depending on order of operations. But it's an option.
3) Defer updating these fields until later. This add some complexity and potential for error (i.e. creating a render node and setting the ID on the component is no longer an atomic operation.)

For now the current implementation is most like option 1. We'll go with it for now and maybe a better solution will present itself later.